test/feature/reverse_routing.feature

```diff
@@ -0,0 +1,31 @@
+Feature: reverse routing
+
+  Background:
+    Given an empty "controllers" directory
```

Comment:
Yes, for those edge-case specs simply delete the directory. 

In general with Cucumber, this is supposed to be non-technical, almost like human-facing documentation that a PM or an RMS executive understands by just glancing over it. Imagine you explain this to a human idiot (like me). When explaining the different aspects of the routes file to me, you wouldn't always start with an empty directory, and then always first create a controllers directory. You would start with a "normal" app, and then tell me "look, when I add 'GET /some/route' to the routes file here, then it calls this controller automatically." After I say "Wow, cool", and ask what happens if the controller doesn't exist, you would simply delete it, boot the app again, and show me that it throws the right exception. The cucumber specs should be on the same level. Simple, casual, end-user friendly walk through of the different functional aspects. 

They should be like the UI. Wrapping the complex technicalities under an intuitive and easy to use UI. They shouldn't sound/feel/be very technical.